hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -269,30 +270,35 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats,
     return true;
   }
 
+  protected HoodieCommitMetadata reconcileCommitMetadata(HoodieTable table, String commitActionType, String instantTime, HoodieCommitMetadata originalMetadata) {
```

Comment:
Can we name it buildCommitMetadata and embed the reconcile logic in. Like this:

```java
HoodieCommitMetadata buildCommitMetadata {
  HoodieCommitMetadata metadata = CommitUtils.buildMetadata(...);
  if (is_old_release) {
    // do the reconcile
  } else {
    return metadata;
  }
}
```

and by default impl, we just return the metadata itself.

Reply:
The `reconcileMetadataForMissingFiles` is being used in other methods as well. I have restructured in a similar way. PTAL.